framework/OutStreams/OutStreamInterface.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 1, 2021
+
+@author: talbpaul
+"""
+import os
+
+import DataObjects
+import Models
+from BaseClasses import BaseInterface
+from utils import InputTypes, InputData
+
+class OutStreamInterface(BaseInterface):
+  """
+    Base class for other OutStream Interfaces (Print, Plot).
+    Not meant to be an interface itself.
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    spec.addParam('dir', descr=r"""Designates the directory in which files should be saved. If not
+        an absolute directory, assumes relative to the WorkingDir. """)
+    spec.addParam('overwrite', param_type=InputTypes.BoolType, descr=r"""If True, then any conflicting
+        existing files will be overwritten.""")
+    spec.addSub(InputData.parameterInputFactory('filename', contentType=InputTypes.StringType,
+        descr=r"""Sets the name to use for the saved file."""))
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'OutStreamInterface'
+    self.overwrite = True       # overwrite existing creations?
+    self.subDirectory = None    # directory to save generated files to
+    self.filename = ''          # target file name
+    self.numberAggregatedOS = 1 # number of aggregated outstreams # no addl info from original OutStream
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+    self.subDirectory = spec.parameterValues.get('dir')
+    self.overwrite = spec.parameterValues.get('overwrite')
+    fname = spec.findFirst('filename')
+    if fname is not None:
+      self.filename = fname
+
+  def initialize(self, stepEntities):
+    """
+      Initialize for a new Step
+      @ In, stepEntities, dict, Entities from the Step
+      @ Out, None
+    """
+    super().initialize()
```

Comment:
I do not see the base classes have this method. Please check it.

Reply:
`BaseInterface` has this method. Incidentally, if it didn't, this line would crash the code.